internal/templates/steps.go

```diff
@@ -38,13 +40,47 @@ var _ Step = (*CommandStep)(nil)
 
 func (s *CommandStep) Run(ctx TemplateContext) error {
 	ctx.Logger.Debug("Running command: %s with args: %s", s.Command, strings.Join(s.Args, " "))
-	cmd := exec.CommandContext(ctx.Context, s.Command, s.Args...)
+
+	timeoutCtx, cancel := context.WithTimeout(ctx.Context, 5*time.Minute)
+	defer cancel()
+	cmd := exec.CommandContext(timeoutCtx, s.Command, s.Args...)
 	cmd.Dir = ctx.ProjectDir
 	cmd.Stdin = nil
+
 	out, err := cmd.CombinedOutput()
 	if err != nil {
+		if s.Command == "uv" && len(s.Args) >= 3 && s.Args[0] == "add" && s.Args[2] == "crewai" {
```

Comment:
would it be better to generalize this and make it work for any module not just crewai?